pkg/webhook/validation/uservalidation.go

```diff
@@ -45,33 +45,34 @@ func ValidateUserForFleetCRD(group string, namespacedName types.NamespacedName,
 	return admission.Allowed(fmt.Sprintf(crdAllowedFormat, userInfo.Username, userInfo.Groups, namespacedName))
 }
 
-// ValidateUserForFleetResource checks to see if user is allowed to modify argued fleet resource.
-func ValidateUserForFleetResource(resKind string, namespacedName types.NamespacedName, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+// ValidateUserForResource checks to see if user is allowed to modify argued resource.
+func ValidateUserForResource(resKind string, namespacedName types.NamespacedName, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
 	if isMasterGroupUserOrWhiteListedUser(whiteListedUsers, userInfo) || isUserAuthenticatedServiceAccount(userInfo) {
 		klog.V(2).InfoS("user in groups is allowed to modify fleet resource", "user", userInfo.Username, "groups", userInfo.Groups, "kind", resKind, "namespacedName", namespacedName)
-		return admission.Allowed(fmt.Sprintf(fleetResourceAllowedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
+		return admission.Allowed(fmt.Sprintf(resourceAllowedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
 	}
 	klog.V(2).InfoS("user in groups is not allowed to modify fleet resource", "user", userInfo.Username, "groups", userInfo.Groups, "kind", resKind, "namespacedName", namespacedName)
-	return admission.Denied(fmt.Sprintf(fleetResourceDeniedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
+	return admission.Denied(fmt.Sprintf(resourceDeniedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
 }
 
-// ValidateMemberClusterUpdate checks to see if user is allowed to update argued member cluster resource.
-func ValidateMemberClusterUpdate(currentMC, oldMC fleetv1alpha1.MemberCluster, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+// ValidateMCOrNSUpdate checks to see if user is allowed to update argued member cluster or namespace resource.
+func ValidateMCOrNSUpdate(currentObj, oldObj client.Object, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+	kind := currentObj.GetObjectKind().GroupVersionKind().Kind
 	response := admission.Allowed(fmt.Sprintf("user %s in groups %v most likely updated read-only field/fields of member cluster resource, so no field/fields will be updated", userInfo.Username, userInfo.Groups))
```

Comment:
why init the response? it is overridden later at several places.

Reply:
https://github.com/Azure/fleet/pull/495/files#r1304893908

Reply:
the go pattern should be return the response inside each if block, not at the end.

Reply:
your  handleNamespace funciton has the right pattern